pydantic/types.py

```diff
@@ -687,6 +704,102 @@ def get_secret_value(self) -> bytes:
         return self._secret_value
 
 
+class PaymentCardBrand(Enum):
+    amex = 'American Express'
+    mastercard = 'Mastercard'
+    visa = 'Visa'
+    other = 'other'
+
+    def __str__(self) -> str:
+        return self.value
+
+
+class PaymentCardNumber(str):
+    """
+    Based on: https://en.wikipedia.org/wiki/Payment_card_number
+    """
+
+    strip_whitespace: ClassVar[bool] = True
+    min_length: ClassVar[int] = 12
+    max_length: ClassVar[int] = 19
+    bin: str
+    last4: str
+    brand: PaymentCardBrand
+
+    def __init__(self, card_number: str):
+        self.bin = card_number[:6]
+        self.last4 = card_number[-4:]
+        self.brand = self.get_brand(card_number)
+
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        yield not_none_validator
+        yield str_validator
+        yield constr_strip_whitespace
+        yield constr_length_validator
+        yield cls.validate_digits
+        yield cls.validate_luhn_check_digit
+        yield cls
```

Comment:
is this required?

Reply:
`PaymentCardNumber.validate_length_for_brand` uses `PaymentCardNumber.brand` which either needs to be set in the constructor or in another validator. In `SecretStr`, this is done in `SecretStr.validate`. It'd be awkward in this case, since one of the three class specific validators would need to call the constructor.

After cleaning it up with @dmontagu's help, I find it to be cleanest implementation, but it's clear when the constructor is being called and converting from `str` to `PaymentCardNumber`.

Reply:
Okay, keep like this.